packages/documentsearch/style/base.css

```diff
@@ -34,11 +34,15 @@
   outline: 0;
 }
 
-.jp-DocumentSearch-overlay button:hover {
+.jp-DocumentSearch-overlay button.jp-DocumentSearch-button-wrapper:disabled {
+  opacity: 0.6;
+}
+
+.jp-DocumentSearch-overlay button:not(:disabled):hover {
```

Comment:
Performance: having a generic element with three pseudo-elements on the right hand side of selector is not great. I don't feel like fiddling with it, but one way to make it work and eliminate performance impact would be making all the buttons within `.jp-DocumentSearch-overlay` use a common class name (in addition to the specific class names they already have). Another solution would be switching to `jp-button` element from `ui-compontents` which is only slightly better than `button` because if user has 1000s native `button`s in their notebook our styles would not cause any performance penalty (of course until they start using `jp-button`).